packages/services/policy/src/environment.ts

```diff
@@ -0,0 +1,116 @@
+import zod from 'zod';
+
+const isNumberString = (input: unknown) => zod.string().regex(/^\d+$/).safeParse(input).success;
+
+const numberFromNumberOrNumberString = (input: unknown): number | undefined => {
+  if (typeof input == 'number') return input;
+  if (isNumberString(input)) return Number(input);
+};
+
+const NumberFromString = zod.preprocess(numberFromNumberOrNumberString, zod.number().min(1));
+
+// treat an empty string (`''`) as undefined
+const emptyString = <T extends zod.ZodType>(input: T) => {
+  return zod.preprocess((value: unknown) => {
+    if (value === '') return undefined;
+    return value;
+  }, input);
+};
+
+const EnvironmentModel = zod.object({
+  PORT: emptyString(NumberFromString.optional()),
+  ENVIRONMENT: emptyString(zod.string().optional()),
+  RELEASE: emptyString(zod.string().optional()),
+});
+
+const SentryModel = zod.union([
+  zod.object({
+    SENTRY: emptyString(zod.literal('0').optional()),
+  }),
+  zod.object({
+    SENTRY: zod.literal('1'),
+    SENTRY_DSN: zod.string(),
+  }),
+]);
+
+const PrometheusModel = zod.object({
+  PROMETHEUS_METRICS: emptyString(zod.union([zod.literal('0'), zod.literal('1')]).optional()),
+  PROMETHEUS_METRICS_LABEL_INSTANCE: emptyString(zod.string().optional()),
+});
+
+const LogModel = zod.object({
+  LOG_LEVEL: emptyString(
+    zod
+      .union([
+        zod.literal('trace'),
+        zod.literal('debug'),
+        zod.literal('info'),
+        zod.literal('warn'),
+        zod.literal('error'),
+        zod.literal('fatal'),
+        zod.literal('silent'),
+      ])
```

Comment:
```suggestion
    zod
      .enum([
        'trace',
        'debug',
        'info',
        'warn',
        'error',
        'fatal',
        'silent',
      ])
```

should be the same but less typed?

Reply:
I'm leaving it as-is, you are right, but it's all over the codebase, so we can fix it in a separate PR.